### DISCUSSION TITLE
Help with setting standard initialization parameters (IE initial temperature)

### DISCUSSION THREAD

**Discussion by ShardulKamatNU** [ansys/pyfluent#1350] [Q&A] [answered]
####################################
NOTE FOR THE FUTURE:
In this original post, 'solver' was the session object created with launch_fluent. later, in the replies, 'session' is the session object
####################################

Hello, I am a relatively new pyfluent user. 
Using pyfluent 0.12.3 on fluent 22.2.0
I have searched for related topics and have found nothing of help
I am using the API reference for version 0.12

Currently, for my use case, I have created a simulation case and data file created in the fluent GUI, which I load using pyfluent, and then I change various attributes before running. The end goal is to do large parametric studies in an automated fashion.

Currently I wish to alter the initial temperature used in standard initialization. However, I cannot find any way to do so. 

Here are the details of my attempts so far, apologies in advance for my lack of understanding of pyfluent and various jargon terms:

solver.solution.initialization() gives the following:

{'fmg_initialize': False,
 'reference_frame': 'relative',
 'set_hybrid_init_options': {'general_settings': {'number_of_iterations': 10,
   'explicit_urf': [1.0, 1.0],
   'reference_frame': 'relative',
   'initial_pressure': False,
   'external_aero': False,
   'const_velocity': False}}}

There is nothing here which can set attributes in standard initialization

The API reference lists the following child:
     solver.solution.initialization.set_defaults()
and the following command
     solver.solution.initialization.list_defaults()
which I guess would be helpful for setting the parameters I want

however, doing  
     solver.solution.initialization.child_names
gives
['fmg_initialize',
 'localized_turb_init',
 'reference_frame',
 'fmg_options',
 'set_hybrid_init_options',
 'patch']

likewise 
     solver.solution.initialization.command_names
gives
['standard_initialize',
 'hybrid_initialize',
 'dpm_reset',
 'lwf_reset',
 'init_flow_statistics',
 'init_acoustics_options']

so it looks like I am missing some commands which are listed in the API reference, I don't know how or why

I have also tried
     solver.tui.solution.initialize.set_defaults() which is listed in the API reference
but I get
     RuntimeError: menu not found
I don't know why this error is showing up. 

So here I am stuck, I don't know how to access the settings for standard initialization using pyfluent
I appreciate any help which can be provided.

**Comment by sravanansys** [comment]
`session.solution.initialization.set_defaults()` gives you a dictionary of initial values to be set for each variable.

`session.solution.initialization.standard_initialize()` lets you use standard initialization instead of hybrid initialization.

![image](https://user-images.githubusercontent.com/99287292/218643021-b09e39d4-ae2b-47fd-a4cd-2b093e6d41b2.png)

**Reply by ShardulKamatNU** [reply]
Hello, thank you for your answer

Assuming that
session = pyfluent.launch_fluent()

when I do 
session.solution.initialization.set_defaults()
I get 
AttributeError: 'initialization' object has no attribute 'set_defaults'

when I do 
solver.solution.initialization.child_names
I get

['fmg_initialize',
'localized_turb_init',
'reference_frame',
'fmg_options',
'set_hybrid_init_options',
'patch']

there is no 'set_defaults' there when it should be there

please see the attached picture. Is there something else that could be wrong? 
![Capture](https://user-images.githubusercontent.com/114320718/218647045-1f5e5ca7-f78d-486d-840b-6d2c83aa51cc.PNG)

**Reply by sravanansys** [reply]
Here is how I am launching the fluent solver session:

![image](https://user-images.githubusercontent.com/99287292/218647396-4bfc9a0d-d9c5-41c8-b025-626ebbac40c0.png)

**Reply by ShardulKamatNU** [reply]
Hello, I have altered my script to use commands as close to how you are using them as I can, while still maintaining those things that are necessary for my script. 

Please see the attached image. I think I am doing things the same way that you are, but I still get
AttributeError: 'initialization' object has no attribute 'set_defaults'
![Capture](https://user-images.githubusercontent.com/114320718/218650948-c162b5b4-2394-4c65-8f61-a046ea0485db.PNG)

**Comment by sravanansys** [comment, marked as answer]
It looks like `set_defaults()` method is not available for version 2022R2. You may have to use the TUI command such as `session.tui.solve.initialize.set_defaults('k',0.001,'omega',100)` followed by `session.solution.initialization.standard_initialize()`

**Reply by ShardulKamatNU** [reply]
Hello, thank you, this has solved my problem.

Originally, I had tried using (among other things)
session.tui.solution.initialize.set_defaults()
which did not work, instead, as you suggested
session.tui.solve.initialize.set_defaults()
worked

Again, my thanks
